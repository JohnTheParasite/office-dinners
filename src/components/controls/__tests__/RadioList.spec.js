import { createLocalVue, shallowMount } from "@vue/test-utils"
import radioList from "@/components/controls/RadioList"
import i18n from "@/i18n"

describe("RadioList render", () => {
  const localVue = createLocalVue()
  localVue.use(i18n)

  let mockOptions = [
    { text: "user.admin", value: 1, selected: false },
    { text: "user.user", value: 2, selected: false }
  ]

  it("should render RadioList element", () => {
    let radioListElement = shallowMount(radioList, { localVue, i18n, propsData: { options: mockOptions } })
    expect(radioListElement).toMatchSnapshot()
  })
})

describe("RadioList with horizontal property", () => {
  const localVue = createLocalVue()
  localVue.use(i18n)

  let mockOptions = [
    { text: "user.admin", value: 1, selected: false },
    { text: "user.user", value: 2, selected: false }
  ]

  it("Orientation property should work", () => {
    let radioListElement = shallowMount(radioList, {
      localVue,
      i18n,
      propsData: { options: mockOptions, orientation: "horizontal" }
    })
    expect(radioListElement).toMatchSnapshot()
  })
})

describe("RadioList with initValue property", () => {
  const localVue = createLocalVue()
  localVue.use(i18n)

  let mockOptions = [
    { text: "user.admin", value: 1, selected: false },
    { text: "user.user", value: 2, selected: false }
  ]

  it("initValue property should work", () => {
    let radioListElement = shallowMount(radioList, { localVue, i18n, propsData: { options: mockOptions, initValue: "2" } })
    expect(radioListElement).toMatchSnapshot()
  })
})
