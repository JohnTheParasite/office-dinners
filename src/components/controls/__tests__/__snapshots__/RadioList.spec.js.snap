// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioList render should render RadioList element 1`] = `<div class="input radio-list form-group"><span class="input-label" style="display: none;"></span> <span class="input-block"><label for="2-1" class="option"><input id="2-1" name="" type="radio" value="1"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>first element</span></label><label for="2-2" class="option"><input id="2-2" name="" type="radio" value="2"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>second element</span></label></span></div>`;

exports[`RadioList with horizontal property Orientation property should work 1`] = `<div class="input radio-list form-group horizontal"><span class="input-label" style="display: none;"></span> <span class="input-block"><label for="4-1" class="option"><input id="4-1" name="" type="radio" value="1"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>user.admin</span></label><label for="4-2" class="option"><input id="4-2" name="" type="radio" value="2"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>user.user</span></label></span></div>`;

exports[`RadioList with initValue property initValue property should work 1`] = `<div class="input radio-list form-group"><span class="input-label" style="display: none;"></span> <span class="input-block"><label for="6-1" class="option"><input id="6-1" name="" type="radio" value="1"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>user.admin</span></label><label for="6-2" class="option"><input id="6-2" name="" type="radio" value="2"> <span class="radio-control"><span class="radio-border"></span> <span class="radio-checked"></span></span> <span>user.user</span></label></span></div>`;
